### PR TITLE
🔑  Added keyByProperty method

### PR DESCRIPTION
**Pull request by nicolasgramlich**

### CODE REVIEW

File: src/interfaces/collection.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export interface ICollection<T> {
   findByProperties(properties: Partial<T>): T;
   forEach(callback: (object: T, index?: number) => void): void;
   mapByProperty<P extends keyof T>(property: P): (T[P])[];
+  keyByProperty<P extends keyof T>(property: P): { [p: string]: T; };
```
Comment (nicolasgramlich) [inline src/interfaces/collection.ts:11]: Typescript allows bracket property-access, no matter if we restrict to `number` or `string` in `[p: string]`. See sample below:

File: test/classes/collection.test.ts [hunk 2]
```diff
@@ -91,6 +91,15 @@ describe('Collection', function () {
     });
   });
 
+  describe('#keyByProperty()', function () {
+    it('should key values', () => {
+      const coll = new Collection([{ a: '1' }, { a: '2' }, { a: '2' }]);
+      const values = coll.keyByProperty('a');
```
Comment (nicolasgramlich) [inline test/classes/collection.test.ts:97]: ... Example:
We can both:
```
values[1].a; // '1'
values['1'].a; // '1'
```

File: src/interfaces/collection.ts (hunk 1 above)
Comment (SylvainEstevez) [reply src/interfaces/collection.ts:11]: I've alway thought that this is a perfect use case for `Map`, which would preserve the value type. WDYT @nicolasgramlich ?

Comment (nicolasgramlich) [reply src/interfaces/collection.ts:11]: Hm, I guess I'm too much of a Lodash fanboy and used to the syntax. 😆 

I actually started with a Map, but decided to not convert from the Lodash object result to a Map.

I'd say 99% of the time the use case is the Lodash way, isn't it?

Maybe we could have both `keyBy` (Map Version) and `keyByProperty` (Lodash version).

Comment (SylvainEstevez) [reply src/interfaces/collection.ts:11]: Alright, alright, let's lodash this!

**Review by SylvainEstevez — APPROVED** [review]

**Comment by nicolasgramlich** [comment]
🔖  `v2.2.0`

**Pull request merged**